### PR TITLE
fix ip6_make_skb argument changes in 5.18

### PR DESCRIPTION
**Pull request by brycekahle**
### What does this PR do?

- Handles a change in the `ip6_make_skb` function arguments in 5.18.
- Fixes an error in the `PT_REGS_PARM9` macro for `arm64`.

### Motivation

https://github.com/torvalds/linux/commit/f37a4cc6bb0ba08c2d9fd7d18a1da87161cbb7f9

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: pkg/network/tracer/offsetguess/tracer.go [hunk 1]
```diff
@@ -576,7 +577,12 @@ func (t *tracerOffsetGuesser) Guess(cfg *config.Config) ([]manager.ConstantEdito
 		What:         uint64(GuessSAddr),
 	}
 
-	if !cfg.CollectIPv6Conns {
+	kv, err := kernel.HostVersion()
+	if err != nil {
+		return nil, err
+	}
+	guessIPv6 := cfg.CollectIPv6Conns && kv < kernel.VersionCode(5, 18, 0)
```
Comment (hmahmood) [inline pkg/network/tracer/offsetguess/tracer.go:584]: I thought this was only for flowi6, i.e. `ip6_make_skb`?

Comment (hmahmood) [reply pkg/network/tracer/offsetguess/tracer.go:584]: Trying to make ci pass for this one I guess?

Comment (brycekahle) [reply pkg/network/tracer/offsetguess/tracer.go:584]: Let me double check if any tests fail without having `flowi6` offsets.

**Comment by akarpz** [comment]
should this have changes for CO-RE as well? I had a draft of that: https://github.com/DataDog/datadog-agent/pull/16072/files#diff-7cfdeaf6b93e5460a293b68f13e31ed8dfed18f9c12e1afb2fdc1b4beb7c31d0

**Comment by brycekahle** [comment]
> should this have changes for CO-RE as well

@akarpz Yeah, I can add that. Since the CO-RE version is currently specific to the runtime environment, which is kernel 5.10, it wasn't an initial focus.

**Review by usamasaqib — APPROVED** [review]

File: pkg/network/tracer/tracer_linux_test.go [hunk 2]
```diff
@@ -1414,6 +1418,10 @@ func TestSendfileRegression(t *testing.T) {
 
 	for _, family := range []network.ConnectionFamily{network.AFINET, network.AFINET6} {
 		t.Run(family.String(), func(t *testing.T) {
+			if family == network.AFINET6 && !isTestIPv6Enabled(cfg) {
```
Comment (hmahmood) [inline pkg/network/tracer/tracer_linux_test.go:1421]: We're skipping even tcp?

Comment (brycekahle) [reply pkg/network/tracer/tracer_linux_test.go:1421]: I've got a separate branch that adds config options for TCPv6 and UDPv6, so we can narrow the impact to just UDPv6.  I'm waiting on CI restoration to open the PR.

**Review by hmahmood — APPROVED** [review]

File: pkg/network/tracer/tracer_linux_test.go (hunk 2 above)
Comment (brycekahle) [reply pkg/network/tracer/tracer_linux_test.go:1421]: #16843

**Review by akarpz — APPROVED** [review]

**Review by richl-dd — APPROVED** [review]
Looks good for WKIT owned files.

**Pull request merged**